fastlane_core/lib/fastlane_core/itunes_transporter.rb

```diff
@@ -800,10 +812,20 @@ def upload(app_id = nil, dir = nil, package_path: nil, asset_path: nil, platform
     # @return (Bool) True if everything worked fine
     # @raise [Deliver::TransporterTransferError] when something went wrong
     #   when transferring
-    def verify(app_id = nil, dir = nil, package_path: nil)
-      raise "Either a combination of app id and directory or a package_path are required" if (app_id.nil? || dir.nil?) && package_path.nil?
+    def verify(app_id = nil, dir = nil, package_path: nil, asset_path: nil, platform: nil)
+      raise "app_id and dir are required or package_path or asset_path is required" if (app_id.nil? || dir.nil?) && package_path.nil? && asset_path.nil?
+
+      force_itmsp = FastlaneCore::Env.truthy?("ITMSTRANSPORTER_FORCE_ITMS_PACKAGE_UPLOAD")
+      can_use_asset_path = Helper.is_mac? && asset_path
 
-      actual_dir = if package_path
+      actual_dir = if can_use_asset_path && !force_itmsp
+                     # The asset gets deleted upon completion so copying to a temp directory
+                     # (with randomized filename, for multibyte-mixed filename upload fails)
+                     new_file_name = "#{SecureRandom.uuid}#{File.extname(asset_path)}"
+                     tmp_asset_path = File.join(Dir.tmpdir, new_file_name)
+                     FileUtils.cp(asset_path, tmp_asset_path)
+                     tmp_asset_path
```

Comment:
You can use `Tempfile.create` to randomize temporary files. https://docs.ruby-lang.org/en/3.2/Tempfile.html

Reply:
Ended up not changing the code for this as I copied it from the `upload` command. As it needs to copy a file to a directory, I couldn't find any easy ways to use `Tempfile` other than having to open and copy the contents of the asset path to the temp path. 

Are you happy with me leaving it as is or do you know of anyway you can get a temp path without creating a path? 🤔

Reply:
I thought again. it's a good situation to use `Dir.mktmpdir` instead. not `Tempfile.create` 🙇 
https://ruby-doc.org/stdlib-2.5.1/libdoc/tmpdir/rdoc/Dir.html

I added suggestions to use this method.
Thanks to this method, we can get rid of the randomize logic.

This tempdir is not automatically deleted, however, it will be deleted on the below line.
https://github.com/fastlane/fastlane/pull/20738/files#diff-e4392686971e0801abe75321cc8a4c90eb2f2c971734d8cc88b0ac5fef5ee12dR860